library/core/src/iter/adapters/step_by.rs

```diff
@@ -40,6 +44,7 @@ impl<I> StepBy<I> {
     /// The `step` that was originally passed to `Iterator::step_by(step)`,
     /// aka `self.step_minus_one + 1`.
     #[inline]
+    #[requires(self.step_minus_one != usize::MAX)]
```

Comment:
```suggestion
```

as this function isn't `unsafe`. We need to use a type invariant instead, see above.